botocore/docs/sharedexample.py

```diff
@@ -0,0 +1,190 @@
+
+        :param section: The section to write to.
+        """
+        section.style.new_paragraph()
+        section.write(example.get('description'))
+        section.style.new_line()
+        self.document_input(section, example, prefix)
+        self.document_output(section, example)
+
+    def document_input(self, section, example, prefix):
+        input_section = section.add_new_section('input')
+        input_section.style.start_codeblock()
+        if prefix is not None:
+            input_section.write(prefix)
+        params = example['input']
+        comments = example.get('comments')
+        if comments:
+            comments = comments.get('input')
+        param_section = input_section.add_new_section('parameters')
+        self._document_params(param_section, params, comments, [])
+        closing_section = input_section.add_new_section('input-close')
+        closing_section.style.new_line()
+        closing_section.style.new_line()
+        closing_section.write('print(response)')
+        closing_section.style.end_codeblock()
+
+    def document_output(self, section, example):
+        output_section = section.add_new_section('output')
+        output_section.writeln('Expected Output:')
+        output_section.style.start_codeblock()
+
+        if 'output' in example:
+            params = example['output']
+        else:
+            params = {}
+
+        # There might not be an output, but we will return metadata anyway
+        params['ResponseMetadata'] = {"...": "..."}
+        comments = example.get('comments')
+        if comments:
+            comments = comments.get('output')
+        self._document_dict(output_section, params, comments, [], True)
+        closing_section = output_section.add_new_section('output-close')
+        closing_section.style.end_codeblock()
+
+    def _document(self, section, value, comments, path):
+        """
+        :param section: The section to add the docs to.
+        :param value: The input / output values representing the parameters that
+                      are included in the example.
+        :param comments: The dictionary containing all the comments to be
+                         applied to the example.
+        :param path: A list describing where the documenter is in traversing the
+                     parameters. This is used to find the equivalent location
+                     in the comments dictionary.
+        """
+        if isinstance(value, dict):
+            self._document_dict(section, value, comments, path)
+        elif isinstance(value, list):
+            self._document_list(section, value, comments, path)
+        elif isinstance(value, numbers.Number):
+            self._document_number(section, value, comments, path)
+        elif isinstance(value, datetime):
+            self._document_datetime(section, value, comments, path)
+        elif isinstance(value, string_types):
+            self._document_str(section, value, comments, path)
```

Comment:
Yeah let's just fallback to string if nothing gets caught in this if statement tree so we do not leave it blank if the value is missed.